File: JavaSamples/src/com/predixion/sample/OutputHelper.java
package com.predixion.sample;
import Interfaces.IPXDataReader;
import Streams.CSVFileHelper;


public class OutputHelper {
	public static void WriteOutputReaderToFile(String outputFileName , IPXDataReader outputReader , char csvSeparator) throws Exception
	{
		String outputDataFile = outputFileName;
		boolean writeHeader = true;
		CSVFileHelper.CreateCSVFileFromDataReader(
				outputReader, 
				outputDataFile, 
				writeHeader, 
				csvSeparator);
		System.out.println(String.format("Output written to file %s", outputFileName));
	}
	
	public static void WriteOutputReaderToConsole(IPXDataReader outputReader) throws Exception
	{
		while (outputReader.Read())
		{
			Object[] values = new Object[outputReader.FieldCount()];
			outputReader.GetValues(values);
			for (int i = 0; i < values.length; i++)
			{
				System.out.println(values[i] + ",");
			}
			System.out.println("\n");
		}
	}
}

File: DotNetSamples/MLSMSample/OutputHelper.cs
using Predixion.Analytics.Interfaces;
using Predixion.MLSMClient.Helpers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MLSMSample
{
    class OutputHelper
    {
    

	public static void WriteOutputReaderToFile(string outputFileName , IPXDataReader outputReader , char csvSeparator)
	{
        String outputDataFile = outputFileName;
		bool writeHeader = true;
		CSVFileHelper.CreateCSVFileFromDataReader(
				outputReader, 
				outputDataFile, 
				writeHeader, 
				csvSeparator);
        Console.WriteLine(string.Format("Output written to file {0}", outputFileName));
	}
	
	public static void WriteOutputReaderToConsole(IPXDataReader outputReader)
	{
		while (outputReader.Read())
		{
			Object[] values = new Object[outputReader.FieldCount];
			outputReader.GetValues(values);
			for (int i = 0; i < values.Length; i++)
			{
				Console.WriteLine(values[i] + ",");
			}
			Console.WriteLine("\n");
		}
	}
    }
}
